mkosi/__init__.py

```diff
@@ -4126,14 +4127,18 @@ def secure_boot_sign(
         return
 
     with mount():
-        for path, _, filenames in os.walk(root / "efi"):
+        base_path = root / "efi" if esp else root / "usr/lib/systemd/boot/efi"
+        for path, _, filenames in os.walk(base_path):
             for i in filenames:
+                p = os.path.join(path, i)
                 if not i.endswith(".efi") and not i.endswith(".EFI"):
                     continue
+                if os.path.exists(f"{p}.signed"):
+                    # Do not overwrite existing signed EFI binaries
```

Comment:
Writing a warning to the user here might be nice, since then there's a clear answer to "why didn't it sign X with my keys?"

Reply:
I wouldn't want, just an informational statement is sufficient here since it's not actually a problem if distros ship presigned binaries

Reply:
Fixed.